check-engines-requirements.js

```diff
@@ -0,0 +1,13 @@
+var semver = require('semver')
+var packageJson = require('./package')
+
+const nodeVersion = packageJson.engines.node
+
+if (!semver.satisfies(process.version, nodeVersion)) {
+  console.log(
+    `Required node version ${nodeVersion} not satisfied with current version ${
+      process.version
+    }.`
+  )
+  process.exit(1)
+}
```

Comment:
We might want the file to be executable and have `#!/usr/bin/env node` at the top.

Reply:
Also, let's make this a hidden file, something like `.check-engine-version`.